### PR TITLE
FORMS-13811 | Adding rendition for hCaptcha

### PR DESCRIPTION
**Pull request by navneet-ag**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by rismehta — CHANGES_REQUESTED** [review]
Please add test cases for this code. You can refer the existing re-captcha implementation

File: packages/react-vanilla-components/src/components/HCaptcha.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  const {
+    label, enabled, visible, required, valid, value,
+    appliedCssClassNames, isError, errorMessage, dispatchChange
+  } = props;
+  const config = getOrElse(props, ['properties', 'fd:captcha', 'config'], {});
+  const bemClass = 'cmp-adaptiveform-hcaptcha';
+
+  if (!config.siteKey) {return null;}
+  return (
+    <div
```
Comment (rismehta) [inline packages/react-vanilla-components/src/components/HCaptcha.tsx:37]: Does this support both visible / invisible behavior ?

Comment (navneet-ag) [reply packages/react-vanilla-components/src/components/HCaptcha.tsx:37]: Invisible is not supported as of now.

File: packages/react-vanilla-components/src/components/HCaptcha.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+//  * distributed under the License is distributed on an “AS IS” BASIS,
+//  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  * See the License for the specific language governing permissions and
+//  * limitations under the License.
+
+//  * The BEM markup is as per the AEM core form components guidelines.
+//  * LINK- https://github.com/adobe/aem-core-forms-components/blob/master/ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/hcaptcha/v1/hcaptcha/recaptcha.html
+//  ******************************************************************************
+
+import React from 'react';
```
Comment (amit-mnnit) [inline packages/react-vanilla-components/src/components/HCaptcha.tsx:20]: @navneet-ag 
can we add the HCaptcha component to the ReCaptcha files (https://github.com/adobe/aem-forms-headless-components/blob/main/packages/react-vanilla-components/src/components/ReCaptcha.tsx)? The only differences are the BEM and library. Everything else remains the same. We can add conditions based on the ':type'.

Comment (navneet-ag) [reply packages/react-vanilla-components/src/components/HCaptcha.tsx:20]: Sure we can do this too.

File: packages/react-vanilla-components/src/components/ReCaptcha.tsx [hunk 3]
```diff
@@ -30,13 +31,16 @@ const ReCAPTCHAComponent = (props: PROPS) => {
     appliedCssClassNames, isError, errorMessage, dispatchChange
   } = props;
   const config = getOrElse(props, ['properties', 'fd:captcha', 'config'], {});
-  const bemClass = 'cmp-adaptiveform-recaptcha';
-
+  const isHCaptcha = function() {
+    return (props[':type'].includes('hcaptcha'));
+  };
+  const bemClass = isHCaptcha() ? 'cmp-adaptiveform-hcaptcha' : 'cmp-adaptiveform-recaptcha';
```
Comment (rismehta) [inline packages/react-vanilla-components/src/components/ReCaptcha.tsx:37]: Why did we implement this way ? Ideally we should have two component types like we have done for core component, this way it is simple was customers to understand and also more discoverable.

**Review by rismehta — CHANGES_REQUESTED** [review]
check comments

File: packages/react-vanilla-components/src/components/ReCaptcha.tsx (hunk 3 above)
Comment (navneet-ag) [reply packages/react-vanilla-components/src/components/ReCaptcha.tsx:37]: This was done after @amit-mnnit 's  request 
https://github.com/adobe/aem-forms-headless-components/pull/136#discussion_r1570045574

Comment (navneet-ag) [reply packages/react-vanilla-components/src/components/ReCaptcha.tsx:37]: I will rever this change.

**Review by rismehta — APPROVED** [review]

**Pull request merged**